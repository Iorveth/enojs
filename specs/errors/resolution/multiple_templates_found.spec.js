const eno = require('../../../eno.js');

const input = `
original: value
original: value

copy < original
`.trim();

describe('resolution.multipleTemplatesFound', () => {

  let error;

  beforeAll(() => {
    try {
      eno.parse(input);
    } catch(err) {
      error = err;
    }
  })

  it(`provides the correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toEqual([[4, 0], [4, 15]]);
  });
});
