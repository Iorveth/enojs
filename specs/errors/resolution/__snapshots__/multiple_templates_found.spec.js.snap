// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolution.multipleTemplatesFound provides correct selection metadata 1`] = `
Array [
  Array [
    3,
    0,
  ],
  Array [
    3,
    15,
  ],
]
`;

exports[`resolution.multipleTemplatesFound provides the correct message 1`] = `
"In line 4 it is not clear which of the elements named 'original' should be copied.

   Line | Content
 >    1 | original: value
 >    2 | original: value
      3 | 
 >    4 | copy < original
"
`;
